packages/ui/src/components/ProjectCard/ProjectCard.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Box, Flex, forwardRef, Grid } from '@chakra-ui/react';
+
+import { statusValueCheck } from '../../helpers/statusValueCheck';
+import { Button } from '../Button/Button';
+import { Typography } from '../Typography/Typography';
+
+type DoneProps = {
+  status: 'done';
+  points: number;
+  pointsMax: number;
+  date?: never;
+};
+
+type NotDoneProps = {
+  status: 'idle' | 'doing' | 'review';
+  date: Date;
+  points?: never;
+  pointsMax?: never;
+};
+
+type CommonProps = {
+  image: string;
+  title: string;
+  url: string;
+};
+
+export type ProjectCardProps = CommonProps & (DoneProps | NotDoneProps);
+
+export const ProjectCard = forwardRef<ProjectCardProps, 'div'>(
+  // disabled for url props
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  ({ status, image, title, url, points, pointsMax, date, ...props }, ref) => {
+    const { statusText, timeStatus, dateOrPoints, isDoing } = statusValueCheck(
+      status,
+      points as number,
+      pointsMax as number,
+      date as Date,
+    );
```

Comment:
🤔 You can pass one argument to the function and type it as `DoneProps | NotDoneProps`. This way there would be no need for casting.